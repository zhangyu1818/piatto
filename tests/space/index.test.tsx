import React from 'react'
import { Space } from 'piatto'
import { render } from '@testing-library/react'
import mountTest from '../shared/mountTest'

describe('Space', () => {
  mountTest(() => (
    <Space>
      <button>1</button>
      <button>2</button>
      <button>3</button>
    </Space>
  ))

  it('should match snapshot', () => {
    const { baseElement } = render(
      <Space>
        <button>1</button>
        <button>2</button>
        <button>3</button>
      </Space>
    )
    expect(baseElement).toMatchSnapshot()
  })

  it('should has gap and gap can change', () => {
    const { baseElement, rerender } = render(
      <Space>
        <button>1</button>
        <button>2</button>
        <button>3</button>
      </Space>
    )
    expect(baseElement.getElementsByClassName('piatto-space-gap-normal').length).toBe(1)
    rerender(
      <Space size="large">
        <button>1</button>
        <button>2</button>
        <button>3</button>
      </Space>
    )
    expect(baseElement.getElementsByClassName('piatto-space-gap-large').length).toBe(1)
  })

  it('should change direction', () => {
    const { baseElement, rerender } = render(
      <Space>
        <button>1</button>
        <button>2</button>
        <button>3</button>
      </Space>
    )
    rerender(
      <Space direction="vertical">
        <button>1</button>
        <button>2</button>
        <button>3</button>
      </Space>
    )
    expect(baseElement.getElementsByClassName('piatto-space-vertical').length).toBe(1)
  })
})
