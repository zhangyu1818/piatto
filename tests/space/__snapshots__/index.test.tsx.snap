// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space should match snapshot 1`] = `
<body>
  <div>
    <div
      class="piatto-space piatto-space-gap-normal piatto-space-horizontal"
    >
      <div
        class="piatto-space-item"
      >
        <button>
          1
        </button>
      </div>
      <div
        class="piatto-space-item"
      >
        <button>
          2
        </button>
      </div>
      <div
        class="piatto-space-item"
      >
        <button>
          3
        </button>
      </div>
    </div>
  </div>
</body>
`;
